watson/watson.py

```diff
@@ -247,6 +248,13 @@ def start(self, project, tags=None):
                 )
             )
 
+        default_tags = self._default_tags(project)
+        if default_tags and not restart:
+            if tags is None:
+                tags = default_tags
+            else:
+                tags = tags + default_tags
```

Comment:
I didn't have time to review the whole PR yet, sorry.

On this one I think that `tags = (tags or []) + default_tags` would be a good option.